### DISCUSSION TITLE
getCurrentUser returning null with Nuxt SSR & Emulator

### DISCUSSION THREAD

**Discussion by timdiacon** [vuejs/vuefire#1404] [Questions and Help]
I have a profile collection where the id of each doc matches the user uid - I am trying to populate a Pinia store with the profile via the following nuxt plugin:

```
export default defineNuxtPlugin(async (nuxtApp) => {
  const user = await getCurrentUser();
  if (user) {
    const db = useFirestore();
    const profile = useDocument(doc(db, "profiles", user.uid));
    const userStore = useUserStore();
    userStore.setProfile(profile);
  } else {
    // No user is signed in.
    console.log("No user is signed in.");
  }
});
```

When I run this locally using a staging firebase project and providing a service worker via the GOOGLE_APPLICATION_CREDENTIALS env var it all works as expected. However when I run it locally using the emulator `getCurrentUser` is returning null on the server but then populating as expected on the client.

I also get the following error in the console log:

`Unknown Error verifying session cookie Failed to determine project ID: Error while making request: getaddrinfo`

I'm unsure whether this is a bug, an issue with SSR / Emulator support or me doing something wrong.

Any advice on whether there is a better way to make the user profile available across the application also welcome :-)

**Comment by stlucasgarcia** [comment]
Same issue here

**Comment by lincolnlemos** [comment]
I had the same problem and I was able to fix the issue by setting up my service account
https://firebase.google.com/docs/admin/setup

These 2 steps were enough for me! I hope it helps
<img width="943" alt="Screenshot 2023-12-11 at 7 17 17 PM" src="https://github.com/vuejs/vuefire/assets/1391743/59c1e90c-4f4b-4844-a167-7678c59a0906">
